panel/src/components/Views/Files/ImageFilePreview.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+	<div
+		:data-has-focus="hasFocus"
+		class="k-default-file-preview k-image-file-preview"
+	>
+		<k-file-preview-thumb :options="options">
+			<k-coords-input
+				:disabled="!focusable"
+				:value="focus"
+				@input="setFocus($event)"
+			>
+				<img v-bind="image" @dragstart.prevent />
+			</k-coords-input>
+		</k-file-preview-thumb>
+
+		<k-file-preview-details :details="details">
+			<div v-if="image.src" class="k-image-file-preview-focus">
+				<dt>{{ $t("file.focus.title") }}</dt>
+				<dd>
+					<k-button
+						v-if="focusable"
+						ref="focus"
+						:icon="focus ? 'cancel-small' : 'preview'"
+						:title="focus ? $t('file.focus.reset') : undefined"
+						size="xs"
+						variant="filled"
+						@click="focus ? setFocus(undefined) : setFocus({ x: 50, y: 50 })"
+					>
+						<template v-if="hasFocus">{{ focus.x }}% {{ focus.y }}%</template>
+						<template v-else>{{ $t("file.focus.placeholder") }}</template>
+					</k-button>
+					<template v-else-if="hasFocus">
+						{{ focus.x }}% {{ focus.y }}%
+					</template>
+					<template v-else>–</template>
+				</dd>
+			</div>
+		</k-file-preview-details>
+	</div>
+</template>
+
+<script>
+/**
+ * File view preview for image files
+ * @since 5.0.0
+ */
+export default {
+	props: {
+		details: {
+			default: () => [],
+			type: Array
+		},
+		focusable: Boolean,
+		image: {
+			default: () => ({}),
+			type: Object
+		},
+		url: String
+	},
+	emits: ["focus"],
+	computed: {
+		focus() {
+			const focus = this.$panel.content.values["focus"];
+
+			if (!focus) {
+				return;
+			}
+
+			const [x, y] = focus.replaceAll("%", "").split(" ");
+
+			return { x: parseFloat(x), y: parseFloat(y) };
+		},
```

Comment:
Could focus be a prop instead, which is provided by the preview?

Reply:
I don't think yet, only once we have changes on the backend.